fluentbit.tf

```diff
@@ -1,40 +1,52 @@
 locals {
+  s3_arn_regex = "^arn:.*:s3:"
+
+  // additional init config files from S3 or files inside a custom image
+  // which are added to the FluentBit container as environment variables, see
+  // https://github.com/aws/aws-for-fluent-bit/tree/develop/use_cases/init-process-for-fluent-bit
+  init_config_files = [
+    for idx, file_or_arn in var.firelens.init_config_files : {
+      name  = format(can(regex(local.s3_arn_regex, file_or_arn)) ? "aws_fluent_bit_init_s3_%s" : "aws_fluent_bit_init_file_%s", idx)
+      value = file_or_arn
+    }
+  ]
+
+  // image tag for the FluentBit container
+  image_tag = length(local.init_config_files) > 0 ? "init-2.32.0.20240122" : "2.32.0"
+
+  // additional init config files ARNs from S3 to be used in an IAM policy for the task role
+  s3_init_file_arns   = [for conf in local.init_config_files : conf.value if can(regex(local.s3_arn_regex, conf.value))]
+  s3_init_bucket_arns = distinct([for arn in local.s3_init_file_arns : split("/", arn)[0]])
+
   // optional FluentBit container for log aggregation
   fluentbit_container_defaults = {
     name                   = var.firelens.container_name
-    image                  = "${data.aws_caller_identity.current.account_id}.dkr.ecr.${data.aws_region.current.name}.amazonaws.com/ecr-public/aws-observability/aws-for-fluent-bit:2.32.0"
+    image                  = "${data.aws_caller_identity.current.account_id}.dkr.ecr.${data.aws_region.current.name}.amazonaws.com/ecr-public/aws-observability/aws-for-fluent-bit:${local.image_tag}"
+    environment            = concat([{ name = "FLB_LOG_LEVEL", value = var.firelens.log_level }], local.init_config_files)
     essential              = true
     mountPoints            = []
     portMappings           = []
     readonlyRootFilesystem = false
+    systemControls         = []
     user                   = startswith(upper(var.operating_system_family), "WINDOWS") ? null : "0:1337"
     volumesFrom            = []
 
-    environment = [
-      // Valid values are: debug, info and error
-      { name = " FLB_LOG_LEVEL", value = "error" }
-    ],
-
     healthCheck = {
       retries = 3
       command = [
         "CMD-SHELL",
-        "curl -s http://localhost:2020/api/v1/uptime | grep uptime_hr | grep -q running"
+        "curl --fail localhost:2020/api/v1/uptime"
```

Comment:
@thisismana this will fail if there isn't a corresponding service config in S3 like described in https://github.com/aws-samples/amazon-ecs-firelens-examples/tree/mainline/examples/fluent-bit/health-check, won't it?

Reply:
works as long as stay with the defaults. But yes, you can disable the health check via configuration and this would fail then